File: 0016-3sum-closest/0016-3sum-closest.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
        Arrays.sort(nums);
        int i=0;
        int st=0;
        int ed=0;
        int ressum=nums[0]+nums[1]+nums[2];
        int mindif=Integer.MAX_VALUE;
        for(int j=0;j<nums.length-2;j++){
            int num=nums[j];
            int left=j+1;
            int right=nums.length-1;
            while(left<right){
                int sum=nums[j]+nums[left]+nums[right];
                if(sum==target){
                    return sum;
                }else if(sum<target){
                    left++;
                }else{
                    right--;
                }
                int dif=Math.abs(sum-target);
                if(dif<mindif){
                    ressum=sum;
                    mindif=dif;
                }
            }
        }
    return ressum; 
    }
}

File: 0016-3sum-closest/0016-3sum-closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        stsum=0
        minsum=0
        diff=10**18
        for i in range(len(nums)-2):
            left=i+1
            right=len(nums)-1
            while left < right:
                sum=nums[i]+nums[left]+nums[right]
                if sum==target:
                    return sum
                elif sum<target:
                    left+=1
                else:
                    right-=1
                dif=abs(target-sum)
                if dif < diff:
                    diff=dif
                    minsum=sum
        return minsum
                


        